chann.go

```diff
@@ -207,13 +207,10 @@ func (ch *Chann[T]) unboundedTerminate() {
 		ch.q = append(ch.q, e)
 	}
 	for len(ch.q) > 0 {
-		select {
-		case ch.out <- ch.q[0]:
-		// The default branch exists because we need guarantee
-		// the loop can terminate. If there is a receiver, the
-		// first case will ways be selected. See #3.
-		default:
-		}
+		// Note if receiver doesn't consume all data that has been sent to input
+		// channel, the `unboundedProcessing` goroutine will leak forever.
+		// Ref: https://github.com/golang-design/chann/issues/3
+		ch.out <- ch.q[0]
```

Comment:
As you documented, if there is no receiver, we will leak a goroutine, and it seems not ideal. Isn't it? Or at least do we need to inform the users in the documentation?

Reply:
Hi @amyangfei can you take a look at this comment? I want to use this library soon

And as for 
> Considering the channel behavior in standard library, if user doesn't receive data from a buffered channel, data in the buffered channel is still kept in memory and has memory leak too, even if the channel is closed.
GC should be able to recycle the unreachable objects when we use built-in channel

Or maybe we can add another function `NewUnbounded[T]() (*Chann[T], func())` that returns a `cleanup` function? like context's `cancel` function, it will remind the user that don't forget to invoke the function.

Reply:
I choose to solve this problem myself :P

https://github.com/lance6716/infchan @amyangfei Will add README later